# 01. Importing Libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [4]:
# Define the project path and Import “orders.csv”

path = r'C:\Users\moein\anaconda3\04-2025 Achievement 4 - Python Fundamentals\02 Data\Original Data'
vars_list = ['order_id', 'user_id', 'order_number', 
             'order_dow', 'order_hour_of_day', 'days_since_prior_order']
df = pd.read_csv(os.path.join(path, 'orders.csv'), usecols=vars_list)

In [5]:
# Print First 5 Rows of df

df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [6]:
# Import “products.csv” as df_prods

df_prods = pd.read_csv(os.path.join(path, 'products.csv'))

In [7]:
# Print Rows from df_prods

# First 20 rows
df_prods.head(20)

# Last 35 rows
df_prods.tail(35)

product_id                                       product_name  \
49658       49654                   Teriyaki Sauce, Sesame, Original   
49659       49655                                        Apple Cider   
49660       49656                         Masada Kosher Pocket Bread   
49661       49657                                  Cabernet Tomatoes   
49662       49658                         Brie with Herbs Foil Wedge   
49663       49659                            Organic Creamed Coconut   
49664       49660                        Professionals Sleek Shampoo   
49665       49661                                              Porto   
49666       49662                       Bacon Cheddar Pretzel Pieces   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   
49668       49664                     Lemon Cayenne Drinking Vinegar   
49669       49665      Super Dark Coconut Ash & Banana Chocolate Bar   
49670       49666                      Ginger Snaps Snacking Cookies   
49671       49667           Enchilada with Spanish Rice & Beans Meal   
49672       49668                     Apple Cinnamon Scented Candles   
49673       49669                                   K Cup Dark Blend   
49674       49670                                Beef Summer Sausage   
49675       49671                               Milk Chocolate Drops   
49676       49672                             Cafe Mocha K-Cup Packs   
49677       49673              Stone Baked Multi Grain Artisan Rolls   
49678       49674            Frozen Greek Yogurt Bars Chocolate Chip   
49679       49675                Cinnamon Dolce Keurig Brewed K Cups   
49680       49676                             Ultra Red Energy Drink   
49681       49677                    Thick & Chunky Sloppy Joe Sauce   
49682       49678                    Large Chicken & Cheese Taquitos   
49683       49679                            Famous Chocolate Wafers   
49684       49680                 All Natural Creamy Caesar Dressing   
49685       49681            Spaghetti with Meatballs and Sauce Meal   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
49658         5             13     4.0  
49659        98              7    10.7  
49660       128              3     7.1  
49661        83              4     8.3  
49662         2             16     3.9  
49663        17             13     3.1  
49664        22             11     6.7  
49665       134              5     8.2  
49666       107             19     3.6  
49667        57             14    10.2  
49668       100             21    13.7  
49669        45             19     6.9  
49670        61             19     5.2  
49671        38              1     6.6  
49672       101             17     5.6  
49673       100             21     4.7  
49674       106             12    19.2  
49675        45             19     3.0  
49676        26              7     6.5  
49677       129              1     5.6  
49678        37              1    11.1  
49679        26              7    14.0  
49680        64              7    14.5  
49681        59             15     8.9  
49682       129              1     3.4  
49683        61             19     6.0  
49684        89             13     4.9  
49685        38              1     6.9  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41     

In [8]:
# Print Column Names

df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [10]:
# Print Shape (Rows & Columns)

df_prods.shape

(49693, 5)

In [14]:
# Max value of “aisle_id”

df_prods['aisle_id'].max()

134

In [15]:
# Data type of “department_id”

df_prods['department_id'].dtype

dtype('int64')

### What’s the maximum value of the “aisle_id” column?
The maximum value of the `aisle_id` column is 134.

### What’s the data type of the “department_id” column?
The `department_id` column is of data type `int64`, which means it stores integer values.